docs/use_cases/advanced_retrieval_augmented_generation.md

```diff
@@ -104,8 +115,10 @@ document_store = InMemoryDocumentStore()
 writer = DocumentWriter(document_store=document_store)
 ```
 
+Indexing organizes the pre-processed data in a structured format that can be quickly and accurately retrieved by the RAG system. This step involves creating an optimized database of tagged and categorized chunks.
+
 ## Retrieval
-Since we use an in-memory database, we import the in-memory retriever. To query our vector database, we need to encode the query texts with the same embedding model used for our document embeddings. The only difference is that for simple text queries, we will be using the `SentenceTransformersTextEmbedder` instead of the document version.
+Because we are using an in-memory database, we import the in-memory retriever. To query our vector database, we need to encode the query texts with the same embedding model used for our document embeddings. The only difference is that for simple text queries, we will be using the `SentenceTransformersTextEmbedder` instead of the document version.

```

Comment:
@xaiguy 
So, are you using both a document version (for document queries) and the simple text version?

Reply:
Yes, it's the same model (has to be, otherwise we couldn't decode the documents), it's just that larger texts are stored as Haystack documents and user queries are usually normal strings and there are different classes (TextEmbedder vs DocumentEmbedder) for both.